Automattic/jetpack
# VideoPress: Shortcode adjustments

Fixes #28045

#### Changes proposed in this Pull Request:

* Fixes the condition used to check if the standalone plugin is installed before overriding Jetpack's shortcodes
* Adds the `muted` attribute to the `video` shortcode override for both Jetpack and standalone VideoPress plugins
* Fixes the `cover` attribute of the VideoPress player on the Jetpack plugin
* Removes the default height of the VideoPress standalone video shortcode, adding a default aspect ratio

The Jetpack version of the shortcode does not support the `height` attribute, whereas the standalone VideoPress version needs it, so the same aspect ratio was introduced to smooth size differences.
The shortcode generated in the standalone plugin's dashboard includes the video's dimensions and represents the intended use.

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Activate VideoPress standalone plugin
* Upload at least one video
* Copy its generated shortcode:
![2022-12-26_18-24-02](https://user-images.githubusercontent.com/8486249/209584045-9ab53010-4616-4e60-8123-f4695af82dbf.png)
![2022-12-26_18-24-21](https://user-images.githubusercontent.com/8486249/209584051-e54ec665-0e52-45ed-b577-96b7aa3220d9.png)
* Disable the VideoPress standalone plugin
* Go to the block editor
* Insert the generated shortcode, with a low width and `cover` set to `true`:
`[videopress btdfDqEs w=320 cover=true]`
* Compare it with the same shortcode, with `cover` set to `false`:
`[videopress btdfDqEs w=320 cover=false]`

Before (With VideoPress installed, but not activated) | Before (Without VideoPress installed) | After
-|-|-
![Screenshot 2022-12-26 at 18-36-14 Shortcode – HelloWord](https://user-images.githubusercontent.com/8486249/209584691-281dbac3-3698-4fbf-8164-cdf5174dda06.png) | ![2022-12-26_18-36-54](https://user-images.githubusercontent.com/8486249/209584713-723c3878-c6a4-4df1-b1ef-a6b9d05f7c48.png) | ![2022-12-26_18-30-56](https://user-images.githubusercontent.com/8486249/209584724-718fa220-e0cb-48be-bb31-f8829bb728d4.png)

* Add a `video` shortcode with the `muted` option:
`[video videopress_guid=btdfDqEs muted=true]`
* Check that the video starts playing muted.

Before | After
-|-
![2022-12-26_18-44-11](https://user-images.githubusercontent.com/8486249/209584954-524db3bd-4f16-4227-bba8-3f5638bf7a46.png) | ![2022-12-26_18-42-59](https://user-images.githubusercontent.com/8486249/209584961-279494ce-1926-416a-a5be-8be0929fa4e5.png)

* Activate the VideoPress standalone plugin
* Check that the previous tests pass on the standalone plugin
* Check that the default height is not a hard value:

Before | After
-|-
![Screenshot 2022-12-26 at 18-36-14 Shortcode – HelloWord](https://user-images.githubusercontent.com/8486249/209585172-8335074d-9b63-4409-bbb4-4391a6af8f6c.png) | ![2022-12-26_18-49-10](https://user-images.githubusercontent.com/8486249/209585184-c03d2a79-0c00-43a6-bbb0-b4f9ad40a731.png)
